.travis.yml

```diff
@@ -6,13 +6,18 @@ python:
 
 sudo: "required"
 
-before_install:
-  - "pip install virtualenv"
-  - "virtualenv venv"
-  - "source venv/bin/activate"
+env:
+  matrix:
+  - TOXENV=foxpuppet GUI=1
 
-install:
-  - "pip install -r requirements.txt "
+before_install:
+  - pip install mozdownload
+  - mozdownload --type=daily
```

Comment:
I would add the branch to make it clear this is a Nightly build, you can also specify a destination filename. You can also use mozinstall and symbolic links to make this the default Firefox version:

```
pip install mozdownload mozinstall
mozdownload --type daily --branch mozilla-central --destination firefox.tar.bz2
mozinstall firefox.tar.bz2
sudo ln -sf $PWD/firefox/firefox /usr/local/bin/firefox
```

I suspect that might help your tests as you'll be using the correct version of Firefox.